pkg/controller/garbagecollector/codec.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package garbagecollector
+
+import (
+	"fmt"
+	"reflect"
+
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/client/typed/dynamic"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+type compressingCodec struct {
+	dynamic.Codec
+}
+
+func NewCompressingCodec() compressingCodec {
+	return compressingCodec{dynamic.Codec{}}
+}
+
+// Decode calls the embedded Codec to decode data. If the decoded object is
+// runtime.Unstructured or runtime.UnstructuredList, Decode only keep the
+// APIVersion, Kind, and ObjectMeta fields.
+func (c compressingCodec) Decode(data []byte, gvk *unversioned.GroupVersionKind, into runtime.Object) (runtime.Object, *unversioned.GroupVersionKind, error) {
+	ret, gvk, err := c.Codec.Decode(data, gvk, into)
```

Comment:
We would inject the shared informer into GC for OpenShift at a code level,
but yes, for resources not compiled into the binary users would need to use
the dynamic client.  We may in the future want to have those servers return
partial responses (objectmeta).

On Thu, Jul 7, 2016 at 1:28 AM, Chao Xu <notifications@github.com> wrote:

> In pkg/controller/garbagecollector/codec.go
> <https://github.com/kubernetes/kubernetes/pull/28480#discussion_r69853004>
> :
>
> > +	"k8s.io/kubernetes/pkg/runtime"
> > +)
> > +
> > +type compressingCodec struct {
> > +	dynamic.Codec
> > +}
> > +
> > +func NewCompressingCodec() compressingCodec {
> > +	return compressingCodec{dynamic.Codec{}}
> > +}
> > +
> > +// Decode calls the embedded Codec to decode data. If the decoded object is
> > +// runtime.Unstructured or runtime.UnstructuredList, Decode only keep the
> > +// APIVersion, Kind, and ObjectMeta fields.
> > +func (c compressingCodec) Decode(data []byte, gvk *unversioned.GroupVersionKind, into runtime.Object) (runtime.Object, *unversioned.GroupVersionKind, error) {
> > +	ret, gvk, err := c.Codec.Decode(data, gvk, into)
>
> A standalone question. @smarterclayton <https://github.com/smarterclayton>
> Apart from thirdparty resources, I think GC cannot use the shared informer
> for openshift objects either. Am I right?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/28480/files/a4f19d24cb9149d4766acab0f77803e63cc5d378#r69853004>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_pzeMWFKqw4dehIX0MEZ6G9IGOS1_ks5qTI6VgaJpZM4JEvyT>
> .
>